middle_end/flambda/simplify/simplify_set_of_closures.ml

```diff
@@ -469,11 +470,15 @@ let simplify_function context ~used_closure_vars ~shareable_constants
     let function_decl = FD.update_code_id function_decl new_code_id in
     let function_type =
       (* We need to use [dacc_after_body] to ensure that all [code_ids] in
-         [params_and_body] are available for the inlining decision code. *)
+         [params_and_body] are available for the inlining decision code.
+         We also provide the latest [cost_metrics] manually as the one present
+         in the denv is the one before simplification.
+      *)
```

Comment:
The first half of this comment is no longer relevant; we don't pass `params_and_body` any more.